src/inmanta/const.py

```diff
@@ -168,20 +168,46 @@ class LogLevel(Enum):
     TRACE = 3
     NOTSET = 0
 
+    @property
+    def to_str(self) -> "StrLogLevel":
+        """
+        Convert an integer formated log level into a string formated log level
+        """
+        return StrLogLevel[self.name]
+
+
+@stable_api
+class StrLogLevel(str, Enum):
+    """
+    Log levels as strings.
+    """
+
+    CRITICAL = "CRITICAL"
+    ERROR = "ERROR"
+    WARNING = "WARNING"
+    INFO = "INFO"
+    DEBUG = "DEBUG"
+    TRACE = "TRACE"
+    NOTSET = "NOTSET"
+
+    @property
+    def to_int(self) -> LogLevel:
+        """
+        Convert a string formated log level into an integer formated log level
+        """
+        return LogLevel[self.name]
+
 
 INMANTA_URN = "urn:inmanta:"
 
 
-class Compilestate(Enum):
+class Compilestate(str, Enum):
```

Comment:
Have you checked that this is not saved to the database? @wouterdb perhaps an annotation for classes that are included in the database could be useful as well, since they need to remain stable too, unless accompanied by a database schema update.

Reply:
As far as I can see, it goes into the database unvalidated, as it comes from the exporter (yikes).
The exporter explicitly uses the string representation.